app/src/main/java/com/it3048/mobiledeviceproject/MainActivity.kt

```diff
@@ -24,31 +25,27 @@ import kotlin.collections.ArrayList
 class MainActivity : AppCompatActivity() {
 
     private var addMeetingForm = AddMeetingForm()
+    private lateinit var binding: MeetingCardBinding
     lateinit var recyclerView: RecyclerView
     lateinit var meetingList:ArrayList<MeetingDAO>
     lateinit var meetingAdapter: MeetingAdapter
-    val inflater = LayoutInflater.from(this)
-    val v = inflater.inflate(R.layout.meeting_card, null)
-    val meetingTitle = v.findViewById<TextView>(R.id.meetingNameTxt)
-    val meetingDate = v.findViewById<TextView>(R.id.meetingDateTxt)
-    val meetingDescription = v.findViewById<TextView>(R.id.meetingDescriptionTxt)
-    val meetingLink = v.findViewById<TextView>(R.id.meetingUrlTxt)
 
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
         setSupportActionBar(findViewById(R.id.toolbar))
+        binding = MeetingCardBinding.inflate(layoutInflater)
 
         /** Set up list */
         meetingList = ArrayList()
         recyclerView = findViewById(R.id.eventsList)
         meetingAdapter = MeetingAdapter(this, meetingList)
         recyclerView.adapter = meetingAdapter
-        val title = meetingTitle.text.toString()
-        val date = meetingDate.text.toString()
-        val description = meetingDescription.text.toString()
-        val link = meetingLink.text.toString()
+        val title = binding.meetingNameTxt.text
+        val date = binding.meetingDateTxt.text
+        val description = binding.meetingDescriptionTxt.text
+        val link = binding.meetingUrlTxt.text
```

Comment:
Used the jetpack viewbinding library to simplify this process and cut out extraneous code. This change also prevents the crash on launch that I was receiving before, which was due to some variables being set before the onCreate function was even called.
Source: https://developer.android.com/topic/libraries/view-binding#activities